### ISSUE TITLE
deprecate usage of ServiceSpreadingPriority and PodFitsPorts

### ISSUE THREAD

**Issue by yastij** [kubernetes/kubernetes#57576]
following slack discussion with @resouer, we might need to really deprecate usage of ServiceSpreadingPriority and PodFitsPorts as names for the according predicate/priority

/sig scheduling
/assign

**Comment by liggitt** [comment]
Details as to why would be appreciated

**Comment by yastij** [comment]
@liggitt - yes sure, these ensure backward compatibility for v 1.0 which we do not support anymore.

**Comment by liggitt** [comment]
They're exposed via the scheduler config file, and can still be used as of the current release.

**Comment by yastij** [comment]
Yes indeed it exposed through config policy, but I still doubt that someone still uses it, also it is a version that we do not support anymore (1.0 is a bit old..). 

Will this break any support policy regarding the supported versions ?

**Comment by liggitt** [comment]
> Yes indeed it exposed through config policy, but I still doubt that someone still uses it, also it is a version that we do not support anymore (1.0 is a bit old..).

Where and when was the predicate announced as deprecated?

> Will this break any support policy regarding the supported versions?

Yes. Generally, a config file that worked in 1.9 should work in 1.10.

What is the downside of leaving that predicate implemented?

**Comment by yastij** [comment]
There's no real downside of leaving it as is, the name is used to register the following predicate : 

https://github.com/kubernetes/kubernetes/blob/572e2764931b9165471e2533a000b2712e44a4c8/plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go#L62-L64

as it said it is only registered for backward compatibility with 1.0

now we've moved to:

https://github.com/kubernetes/kubernetes/blob/572e2764931b9165471e2533a000b2712e44a4c8/plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go#L65-L68

@liggitt - what is usually the timeline to deprecate user-facing things ? 

also cc @resouer WDYT ?

**Comment by yastij** [comment]
cc @bsalamat

**Comment by k82cn** [comment]
if we deprecated and removed them in future, can we achieve the same behaviour by other features or predicates/priorities? If not, what's our suggestion to the cases, ignore it?

Here's some input for those two predicates/priorities:
1. Regarding ServiceSpreadingPriority, can we use [controllerOf](https://groups.google.com/forum/#!topic/kubernetes-dev/KCAwOKy4WAQ) instead of watching RC?
2. Regarding PodFitsPorts, prefer to make it mandatory (https://github.com/kubernetes/kubernetes/issues/50516)

**Comment by bsalamat** [comment]
PodFitsHostPorts covers PodFitPorts, but ServiceSpreadingPriority does not have a replacement and I think should be kept or replaced by another function.
Speaking of deprecation, we also have Service Affinity predicate which is not enabled by default. I am not sure if it is being used. I think it should be removed from the the main scheduler and go to an extender if anyone needs it.

**Comment by k82cn** [comment]
> PodFitsHostPorts covers PodFitPorts

Seems it's only a "name" deprecation (both point to `predicates.PodFitsHostPorts`), I'm ok with that.

> Service Affinity predicate

+1, we can not put every policy in upstream :).
